tests/test_device.py

```diff
@@ -71,15 +71,16 @@ def test_query_gid(self):
                 ctx.query_gid(port_num=1, index=0)
 
     @staticmethod
-    def verify_device_attr(attr):
+    def verify_device_attr(attr, device):
         """
         Helper method that verifies correctness of some members of DeviceAttr
         object.
         :param attr: A DeviceAttr object
         :return: None
         """
-        assert attr.node_guid != 0
-        assert attr.sys_image_guid != 0
+        if device.node_type != 'Unknown':
```

Comment:
do you mean "IBV_NODE_UNKNOWN" ?
node_type is not a string (it's just printed as a string using an internal translation function)

Reply:
Ack